rpcs3/Emu/RSX/RSXFIFO.cpp

```diff
@@ -31,15 +31,23 @@ namespace rsx
 		}
 
 		template <bool full>
-		u32 FIFO_control::read_put()
+		inline u32 FIFO_control::read_put()
 		{
 			if constexpr (!full)
 			{
 				return m_ctrl->put & ~3;
 			}
 			else
 			{
-				return m_ctrl->put.and_fetch(~3);
+				u32 put = m_ctrl->put;
+				if (put & 3)
```

Comment:
```suggestion
				if (UNLIKELY(put & 3))
```

Reply:
rpcs3 has LIKELY and UNLIKELY wrappers around __builtin_expect

Reply:
Or better yet, invert the condition to avoid the unnecessary branch. Probably doesn't matter for modern CPUs but its still good practice.